### PR TITLE
Add EIP: NFT Redeemables

### PR DESCRIPTION
**Pull request by ryanio**
Discussion thread: https://ethereum-magicians.org/t/erc-7498-nft-redeemables/15485

### CODE REVIEW

**Comment by abcoathup** [comment]
EIP/ERC numbering changes to sequential from 7500 and is no longer the PR number: https://github.com/ethereum/EIPs/issues/6990#issuecomment-1648535814

This PR was opened 3 minutes after #7499 so has the appearance of number sniping but as there is a spec I am not manually renumbering.

File: EIPS/eip-7501.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
+---
+eip: 7501
```
Comment (SamWilsn) [inline EIPS/eip-7501.md:2]:
```suggestion
eip: 7498
```

**Review by SamWilsn — CHANGES_REQUESTED** [review]

File: EIPS/eip-7498.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The token MUST have the following interface and MUST return `true` for [ERC-165](./eip-165.md) supportsInterface for `0x12345678`, the 4 byte interfaceId of the below.
+
+```solidity
+interface IERC7501 {
+  /* Events */
+  event CampaignUpdated(uint256 indexed campaignId, CampaignParams params, string URI);
```
Comment (0xanders) [inline EIPS/eip-7498.md:36]: suggestion: change "event CampaignUpdated(uint256 indexed campaignId, CampaignParams params, string URI);" to "event CampaignUpdated(uint256 campaignId, CampaignParams params, string URI); for saving gas"

File: EIPS/eip-7498.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  event Redemption(uint256 indexed campaignId, bytes32 redemptionHash, uint256[] tokenIds, address redeemedBy);
```
Comment (0xanders) [inline EIPS/eip-7498.md:37]: suggestion: change "event Redemption(uint256 indexed campaignId, bytes32 redemptionHash, uint256[] tokenIds, address redeemedBy);"  to "event Redemption(uint256 campaignId, bytes32 redemptionHash, uint256[] tokenIds, address redeemedBy);" for saving gas.

File: EIPS/eip-7498.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+- discovery: events and getters that provide information about the requirements of a redemption campaign
+- onchain: token mints with context of items spent
+- offchain: the ability to associate with ecommerce orders (through `redemptionHash`)
+- trait redemptions: improving the burn-to-redeem experience with E RC-7496 <!-- [E RC-7496] (./e ip-7496.md) --> Dynamic Traits.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The token MUST have the following interface and MUST return `true` for [ERC-165](./eip-165.md) supportsInterface for `0x12345678`, the 4 byte interfaceId of the below.
```
Comment (0xanders) [inline EIPS/eip-7498.md:31]: suggestion: change"0x12345678" to the value of    `type(IERC7501).interfaceId`

**Review by 0xanders — CHANGES_REQUESTED** [review]

File: EIPS/eip-7498.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+    bytes32 substandardValue;
+  }
+
+  /* Getters */
+  function getCampaign(uint256 campaignId) external view returns (CampaignParams memory params, string memory uri, uint256 totalRedemptions);
+
+  /* Setters */
+  function createCampaign(CampaignParams calldata params, string calldata uri) external returns (uint256 campaignId);
+  function updateCampaign(uint256 campaignId, CampaignParams calldata params, string calldata uri) external;
+  function redeem(uint256[] calldata tokenIds, bytes calldata extraData) external;
```
Comment (ArchieR7) [inline EIPS/eip-7498.md:64]: suggestion:
You can consider based on ERC6672 doing the redemption.

File: EIPS/eip-7498.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+    },
+    "description": {
+      "type": "string",
+      "description": "A one-line summary of the redeemable. Markdown is not supported."
+    },
+    "details": {
+      "type": "string",
+      "description": "A multi-line or multi-paragraph description of the details of the redeemable. Markdown is supported."
+    },
+    "imageUrls": {
```
Comment (nickjuntilla) [inline EIPS/eip-7498.md:185]: Why not include an "image" key so these NFTs don't break other places where NFTs are displayed? The image key can then be used as a thumbnail. The metadata format URI should be a superset of normal NFT metadata.

File: EIPS/eip-7498.md (hunk 4 above)
Comment (ryanio) [reply EIPS/eip-7498.md:31]: this was a placeholder until the spec is finalized to provide the actual 4 byte interfaceId for quick reference

File: EIPS/eip-7498.md (hunk 6 above)
Comment (ryanio) [reply EIPS/eip-7498.md:185]: ah yes it should be an array thanks, will fix. the first image should be used as the thumbnail, or can rotate between the multiple.

> Why not include an "image" key so these NFTs don't break other places where NFTs are displayed?

i don't really understand what you mean here, could you elaborate more?

File: EIPS/eip-7498.md (hunk 2 above)
Comment (ryanio) [reply EIPS/eip-7498.md:36]: having the indexed param is useful for querying eth_getLogs based on the campaignId, but will consider removing. what do others think?

File: EIPS/eip-7498.md (hunk 3 above)
Comment (ryanio) [reply EIPS/eip-7498.md:37]: having the indexed param is useful for querying eth_getLogs based on the campaignId, but will consider removing. what do others think?

File: EIPS/eip-7498.md (hunk 5 above)
Comment (ryanio) [reply EIPS/eip-7498.md]: unfortunately we need more functionality than what 6672 provides

File: EIPS/eip-7498.md (hunk 6 above)
Comment (ryanio) [reply EIPS/eip-7498.md:200]: fixed, it's specified as an array now

File: EIPS/eip-7498.md (hunk 2 above)
Comment (ryanio) [reply EIPS/eip-7498.md:36]: in discussion with the rest of my team, although there is a little gas savings we think it's worth it to keep indexed on campaignId

File: EIPS/eip-7498.md (hunk 3 above)
Comment (ryanio) [reply EIPS/eip-7498.md]: in discussion with the rest of my team, although there is a little gas savings we think it's worth it to keep indexed

File: assets/eip-7498/RedeemablesStructs.sol [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+import {OfferItem, ConsiderationItem, SpentItem} from "seaport-types/src/lib/ConsiderationStructs.sol";
+
+struct CampaignParams {
+    uint32 startTime;
+    uint32 endTime;
+    uint32 maxCampaignRedemptions;
+    address manager;
+    address signer;
+    OfferItem[] offer;
```
[the remark this answers is not in the document]
Comment (ryanio) [reply assets/eip-7498/RedeemablesStructs.sol:16]: updated, thx

File: EIPS/eip-7498.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+This specification introduces a new interface that extends [ERC-721](./eip-721.md) and [ERC-1155](./eip-1155.md) to enable the discovery and use of onchain and offchain redeemables for NFTs.
+
+## Motivation
+
+Creators frequently use NFTs to create redeemable entitlements for digital and physical goods. However, without a standard interface, it is challenging for users and apps to discover and interact with these NFTs in a predictable and standard way. This standard aims to encompass enabling broad functionality for:
+
+- discovery: events and getters that provide information about the requirements of a redemption campaign
+- onchain: token mints with context of items spent
+- offchain: the ability to associate with ecommerce orders (through `redemptionHash`)
+- trait redemptions: improving the burn-to-redeem experience with E RC-7496 <!-- [E RC-7496] (./e ip-7496.md) --> Dynamic Traits.
```
Comment (g11tech) [inline EIPS/eip-7498.md:25]: i am assuming you are waiting for 7496 draft merge to fix this link?

**Review by g11tech — COMMENTED** [review]
looks good, minor comment

**Comment by SamWilsn** [comment]
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.

**Pull request closed without merging**